src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy fruitTransactionStrategy) {
+        this.operationStrategy = fruitTransactionStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruits) {
+        fruits.forEach((f -> operationStrategy
+                .getOperationStrategy(f.getOperation())
+                .getOperation(f.getFruit(), f.getQuantity())));
```

Comment:
The method `getOperation` is used to perform an operation, which might be misleading as 'get' typically implies retrieval rather than execution. Consider renaming this method to something more indicative of its action, such as `executeOperation` or `performOperation`, to improve code readability.